src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -40,16 +42,26 @@ public class DiagnosticPipeline : IDisposable
             Requires.NotNull(sinks, nameof(sinks));
             Requires.Argument(sinks.Count > 0, nameof(sinks), "There must be at least one sink");
 
+            this.eventsInProgress = 0;
+
             this.pipelineConfiguration = pipelineConfiguration ?? new DiagnosticPipelineConfiguration();
 
+            // An estimatie how many batches of events to allow inside the pipeline.
+            // We want to be able to process full buffer of events, but also have enough batches in play in case of high concurrency.
+            int MaxNumberOfBatchesInProgress = Math.Max(
+                5 * this.pipelineConfiguration.MaxConcurrency,
```

Comment:
Yes, just a guess for reasonable size of the input buffer for the blocks that handler event batches.